hubploy/auth.py

```diff
@@ -69,49 +70,77 @@ def registry_auth_gcloud(deployment, project, service_key):
+            # Get credentials from standard location
+            service_key_path = os.path.join(
+                'deployments', deployment, 'secrets', service_key
+            )
+
+            if not os.path.isfile(service_key_path):
+                raise FileNotFoundError(
+                    f'The service_key file {service_key_path} does not exist')
+
+            original_credential_file_loc = os.environ.get("AWS_SHARED_CREDENTIALS_FILE", None)
+
+            # Set env variable for credential file location
+            os.environ["AWS_SHARED_CREDENTIALS_FILE"] = service_key_path
+
+            # Requires amazon-ecr-credential-helper to already be installed
+            # this adds necessary line to authenticate docker with ecr
+            docker_config_dir = os.path.expanduser('~/.docker')
+            os.makedirs(docker_config_dir, exist_ok=True)
+            docker_config = os.path.join(docker_config_dir, 'config.json')
+            if os.path.exists(docker_config):
+                with open(docker_config, 'r') as f:
+                    config = json.load(f)
+            else:
+                config = {}
 
-    original_credential_file_loc = os.environ.get("AWS_SHARED_CREDENTIALS_FILE", None)
+            registry = f'{project}.dkr.ecr.{zone}.amazonaws.com'
+            config.setdefault('credHelpers', {})[registry] = 'ecr-login'
+            with open(docker_config, 'w') as f:
+                json.dump(config, f)
 
-    try:
-        # Set env variable for credential file location
-        os.environ["AWS_SHARED_CREDENTIALS_FILE"] = service_key_path
-
-        registry = f'{project}.dkr.ecr.{zone}.amazonaws.com'
-        # Requires amazon-ecr-credential-helper to already be installed
-        # this adds necessary line to authenticate docker with ecr
-        docker_config_dir = os.path.expanduser('~/.docker')
-        os.makedirs(docker_config_dir, exist_ok=True)
-        docker_config = os.path.join(docker_config_dir, 'config.json')
-        if os.path.exists(docker_config):
-            with open(docker_config, 'r') as f:
-                config = json.load(f)
-        else:
-            config = {}
-
-        config.setdefault('credHelpers', {})[registry] = 'ecr-login'
-        with open(docker_config, 'w') as f:
-            json.dump(config, f)
+        if role:
+            # this doesn't come back in the sts client response
+            role_session_name = 'registry'
+
+            sts_client = boto3.client('sts')
+            assumed_role_object = sts_client.assume_role(
+                RoleArn = role,
+                RoleSessionName = role_session_name
+            )
+
+            creds = assumed_role_object['Credentials']
+            os.environ['AWS_ACCESS_KEY_ID'] = creds['AccessKeyId']
+            os.environ['AWS_SECRET_ACCESS_KEY'] = creds['SecretAccessKey']
+            os.environ['AWS_SESSION_TOKEN'] = creds['SessionToken']
+            #os.environ['AWS_ROLE_SESSION_NAME'] = role_session_name
```

Comment:
Does this need to be set?

Reply:
No.  I thought it did at first, but it doesn't.  I'll remove it.